src/sync/sync.nim

```diff
@@ -0,0 +1,238 @@
+import std/[json, os, sequtils, sets, strformat, strutils]
```

Comment:
This file is quite big already. Do you plan on splitting it up into different files later?

Reply:
Yes, probably.

I don't like that currently there's a special file for `update_tests.nim` with everything else in `sync.nim`. It's probably better to have e.g. `sync_docs.nim`, `sync_filepaths.nim` `sync_metadata.nim`, and `sync_tests.nim`.

Reply:
Agreed. That sounds much better!

Reply:
Done. Some improvements still possible, but it's better than it was.

Reply:
Absolutely! It looks much better.